substrate-node/pallets/pallet-smart-contract/src/storage_state/v9.rs

```diff
@@ -0,0 +1,978 @@
+                match contract.contract_type {
+                    types::ContractData::NodeContract(node_contract) => {
+                        let contract_resource = NodeContractResources::<T>::get(contract_id);
+                        r.saturating_inc();
+                        if node_contract.public_ips > 0 || !contract_resource.used.is_empty() {
+                            if contract_id_count[contract_id as usize] == 1 {
+                                new_contract_ids.push(contract.contract_id);
+                            }
+                            contract_id_count[contract_id as usize].saturating_dec();
+                        }
+                    }
+                    _ => {
+                        if contract_id_count[contract_id as usize] == 1 {
+                            new_contract_ids.push(contract.contract_id);
+                        }
+                        contract_id_count[contract_id as usize].saturating_dec();
+                    }
+                };
+            }
+            r.saturating_inc();
+        }
+        ContractsToBillAt::<T>::insert(index, new_contract_ids);
+        w.saturating_inc();
+    }
+
+    debug!(
+        "✨  Smart Contract pallet {:?} cleaning ContractsToBillAt storage map END",
+        PalletVersion::<T>::get()
+    );
+
+    T::DbWeight::get().reads_writes(r.saturating_add(2), w)
+}
+
+// ActiveNodeContracts
+pub fn clean_active_node_contracts<T: Config>() -> frame_support::weights::Weight {
+    debug!(
+        "🧼  Smart Contract pallet {:?} cleaning ActiveNodeContracts storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    let mut r = 0u64;
+    let mut w = 0u64;
+
+    let active_node_contracts = ActiveNodeContracts::<T>::iter().collect::<Vec<_>>();
+    r.saturating_accrue(active_node_contracts.len() as u64);
+
+    for (node_id, mut contract_ids) in active_node_contracts {
+        if pallet_tfgrid::Nodes::<T>::get(node_id).is_none() {
+            ActiveNodeContracts::<T>::remove(node_id);
+            w.saturating_inc();
+        } else {
+            contract_ids.retain(|contract_id| { 
+                r.saturating_inc();
+                Contracts::<T>::get(contract_id).is_some() 
+            });
+            ActiveNodeContracts::<T>::insert(node_id, contract_ids);
+            w.saturating_inc();
+        }
+        r.saturating_inc();
+    }
+
+    debug!(
+        "✨  Smart Contract pallet {:?} cleaning ActiveNodeContracts storage map END",
+        PalletVersion::<T>::get()
+    );
+
+    T::DbWeight::get().reads_writes(r.saturating_add(2), w)
+}
+
+// ActiveRentContractForNode
+pub fn clean_active_rent_contract_for_node<T: Config>() -> frame_support::weights::Weight {
+    debug!(
+        "🧼  Smart Contract pallet {:?} cleaning ActiveRentContractForNode storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    let mut r = 0u64;
+    let mut w = 0u64;
+
+    let to_remove: Vec<_> = ActiveRentContractForNode::<T>::iter()
```

Comment:
We can do translate here also no? To avoid the second iteration...

Reply:
same as above https://github.com/threefoldtech/tfchain/pull/659/files/cc15fc61e7ef318f015e7434c1a88928a12833dc#r1201405720

Here I use 2 loops to avoid copying storage

Reply:
Will mutate work here?